parsing_2.py

```diff
@@ -85,13 +85,38 @@ def main():
             Redirect_count += 1
         elif line.status_code[0] == "4":
             Error_count += 1
-   
+
+
+    #Dictionary to store requests and how often they appear
+    requests_dict = {}
+
+    #basically copied from the example in general
+    #iterates through every request, if it exists it adds one, if not the request is added
+    #to the dictionary
+    for line in log_lines:
+        if line.request in requests_dict:
+            requests_dict[line.request] +=1
+        else:
+            requests_dict[line.request] = 1
+
+    #goes through the dictionary. If the number of requests is higher than max_requests,
+    #that number is stored in max_requests and the name in max_requests_name
+    max_requests = 0
+    max_requests_name = 'start'
+    for request in requests_dict:
+        if requests_dict[request] > max_requests:
+            max_requests = requests_dict[request]
+            max_requests_name = request
+
+
+    total_responses = len(log_lines)
     print(f"Between {first_date_str} and {last_date_str}, there were {len(log_lines)} requests made to our website")
     print(f"In the last six months ({last_six_months_str} - {last_date_str}), there were {len(last_six_months)} requests made to our website")
     print("Total number of redirects:", Redirect_count)
     print("Percentage of redirect request: {0:.1%}".format(Redirect_count/total_responses))
     print("Total number of Errors:", Error_count)
     print("Percentage of error request: {0:.1%}".format(Error_count/total_responses))
+    print("The most requested file: {max_requests_name}")
```

Comment:
I missed something; this should be
```python
    print(f"The most requested file: {max_requests_name}")
```
(note the `f` before the string, which enables string interpolation)